editor/src/components/editor/store/vscode-changes.ts

```diff
@@ -209,12 +209,36 @@ export interface AccumulatedVSCodeChanges {
   selectedChanged: SelectedElementChanged | null
 }
 
+function combineFileChanges(
+  first: Array<ProjectChange>,
+  second: Array<ProjectChange>,
+): Array<ProjectChange> {
+  let writeFilePathsSeen: Set<string> = new Set()
+  let reversedResult: Array<ProjectChange> = []
+  fastForEach([...first, ...second].reverse(), (change) => {
+    if (change.type === 'WRITE_PROJECT_FILE') {
+      if (writeFilePathsSeen.has(change.fullPath)) {
+        // We only want the last version of each file
+        return
```

Comment:
Doesn't this just exit the whole thing regardless of there being other file changes?

Reply:
This just exits this iteration of the for loop (because this is inside the `fastForEach`'s anonymous function). If it were in a regular `for( ... of ...)` then it would exit the `combineFileChanges`

Reply:
But that's what I'm saying, if there's a change for file `A` and then two changes for file `B`, this will result in the change for file `A` being lost surely?

Reply:
Oh no, ignore me, this is being run per-file so it's not an issue.